.github/workflows/rust.yml

```diff
@@ -0,0 +1,42 @@
+name: Rust
+
+on:
+  push:
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Format code
+      run: cargo fmt
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install lld
+      run: |
+        sudo apt install lld
+        sudo apt install libunistring-dev
+        sudo apt install libudev-dev
+    - name: Install Seastar
+      run: |
+        cd ..
+        git clone https://github.com/scylladb/seastar.git
+        cd seastar/
+        sudo ./install-dependencies.sh
+        ./configure.py --mode=release --prefix=/usr/local --without-tests --without-demos --without-apps
+        ninja -C build/release
+        sudo ninja -C build/release install
+        export PKG_CONFIG_PATH="/usr/local/seastar/build/release/:$PKG_CONFIG_PATH"
+    - name: Build
+      run: RUSTFLAGS="-C link-arg=-fuse-ld=lld" cargo build
+    - name: Run linter
+      run: cargo clippy
```

Comment:
Both scylla-rust-driver and cpp-rust-driver use `--verbose` flag for clippy, let's do the same.
scylla-rust-driver additionally uses `--examples --tests` - I hope we are going to have some examples and tests, so let's use those flags too.
cpp-rust-driver also has `-- -D warnings`: https://github.com/scylladb/cpp-rust-driver/blob/9999af005726b9d47d1837536e58d463ec63da63/.github/workflows/clippy.yml#L19 . I think that's a good idea - that way we'll make sure clippy output is clear, so it will be easier to use locally.

Reply:
I think there might be a problem with `-- -D warnings`. Currently, `build.rs` produces a warning - `pkg-config-rs` is unable to parse the `fmt` dependency as it is specified in seastar's pkgconfig file.

If we want to check for warnings in CI (which I agree is a good idea) then we should either fix or slience this warning. At the moment, I don't have a suggestion how to do either of those.